### ISSUE TITLE
[core] Expose DashboardSidebarSubNavigation at slotprops

### ISSUE THREAD

**Issue by H01001000** [mui/toolpad#4269]
### Summary

Expose DashboardSidebarSubNavigation as slot and slotprops, allow using non hardcoded (shared/link component) for sidebar

### Examples

```tsx
import Link from "next/link"
<DashboardLayout slotProps={{sidebarSubNavigation: {LinkComponent: Link} }}>
...
```

### Motivation

Currently when navigating between the page with sidebar, it completely refetch(?) the page and causes state loss (if you collapse the sidebar, it open after click link). 

**Search keywords**: Link, sidebar, dashboard layout

**Comment by apedroferreira** [comment]
Maybe you were following the first example I had written for Vite & React Router? It did have a sidebar "bug" like that which has been updated/fixed since.
React Router integration will be released soon with v0.8.0 of `@toolpad/core`, let us know if you still have any issues then!

**Comment by H01001000** [comment]
Yea I wasn’t paying attention using the default app provider for next js
